src/components/TruncatedUsername/TruncatedUsername.tsx

```diff
@@ -0,0 +1,28 @@
+import { Label } from '@patternfly/react-core';
+
+import { TooltipWrapper } from 'components/TooltipWrapper/TooltipWrapper';
+
+interface ITruncatedTextProps {
+  text: string;
+  length?: number;
+}
+
+/**
+ * The username component that will display a truncated username.
+ *
+ * @param text - The raw text of the username to be truncated
+ * @param length - The target length for usernames that are too long, by default 20
+ */
+export const TruncatedUsername = ({ text, length = 20 }: ITruncatedTextProps) => {
+  const removeKeyword = 'service-account-';
+  let resultText = text.replace(removeKeyword, '');
```

Comment:
I noticed 2 things:
1) much simpler syntax can be used (for `substring` - any argument value that is less than 0 or greater than str.length is treated as if it were 0 and str.length, respectively)
2) `length - 1` is wrong, only `length` should be used, otherwise it's not target length (see your TSDoc)

```tsx
const resultText = text.replace(removeKeyword, '').substring(0, length);

return ...
```

Reply:
done